components/skus/browser/rs/cxx/src/errors.rs

```diff
@@ -133,3 +133,7 @@ impl From<ffi::SkusResult> for skus::errors::InternalError {
         }
     }
 }
+
+pub fn result_to_string(result: &ffi::SkusResult) -> String {
+    <ffi::SkusResult as Into<skus::errors::InternalError>>::into(*result).to_string()
```

Comment:
I'd have a slight preference for something like this, for readability
```rust
let internal_error: skus::errors::InternalError = result.into();
internal_error.to_string()
```

Reply:
This gave me a compilation error:

```> autoninja -C /home/nick/Code/brave-browser/src/out/Component brave -k 1
ninja: Entering directory `/home/nick/Code/brave-browser/src/out/Component'
[2/6] RUST libskus_cxx.rlib
FAILED: libskus_cxx.rlib
"python3" "../../build/rust/rustc_wrapper.py" --rustc=../../third_party/rust-toolchain/bin/rustc --depfile=./libskus_cxx.rlib.d --rsp=./libskus_cxx.rlib.rsp -- -Clinker="../../third_party/llvm-build/Release+Asserts/bin/clang++" --crate-name skus_cxx ../../brave/components/skus/browser/rs/cxx/src/lib.rs --crate-type rlib --edition=2021 -Dunsafe_op_in_unsafe_fn -Dwarnings -Cforce-unwind-tables=yes -Crelocation-model=pic -Coverflow-checks=on -Cdefault-linker-libraries -Zdep-info-omit-d-target -Zmacro-backtrace -Zremap-cwd-prefix=. --color=always --target=x86_64-unknown-linux-gnu -Cembed-bitcode=no -Cpanic=abort -Zpanic_abort_tests --cfg cr_rustc_revision=\"c0583a0221c28f01a7bdbc01464baf22f7c6d4a8-1-llvmorg-18-init-4631-gd50b56d1\" -Copt-level=3 --sysroot=local_rustc_sysroot --emit=dep-info=./libskus_cxx.rlib.d,link -o ./libskus_cxx.rlib LDFLAGS RUSTENV OUT_DIR=/home/nick/Code/brave-browser/src/out/Component/gen/brave/components/skus/browser/rs/cxx/rust_lib
error[E0277]: the trait bound `InternalError: From<&SkusResult>` is not satisfied
   --> ../../brave/components/skus/browser/rs/cxx/src/errors.rs:138:62
    |
138 |     let internal_error: skus::errors::InternalError = result.into();
    |                                                              ^^^^ the trait `From<&SkusResult>` is not implemented for `InternalError`
    |
    = note: required for `&SkusResult` to implement `Into<InternalError>`
help: consider dereferencing here
    |
138 |     let internal_error: skus::errors::InternalError = (*result).into();
    |                                                       ++      +

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
ninja: build stopped: subcommand failed.
null
null
```

Lmk, I can change to
```
pub fn result_to_string(result: &ffi::SkusResult) -> String {
    let internal_error: skus::errors::InternalError = (*result).into();
    internal_error.to_string()
}
```
which worked.

Reply:
But let me know over here https://github.com/brave/brave-core/pull/20800